blog/2020-10-06-monero-bitcoin.md

```diff
@@ -0,0 +1,190 @@
+In our effort to connect all the blockchains without adding yet another, we have turned our attentions to Monero.
+Similar to Grin, Monero is a public ledger in which the source, destination and amount of a transaction remain hidden to observers.
+The team has been interested in Monero for a long time, but until recently we thought that atomic swaps involving it were unfeasible for a few reasons:
+
+1. Timelocks are not supported, so one cannot construct spend conditions contingent on time.
+2. It lacks a scripting language altogether, meaning that the usual mechanism of hashlocks is unavailable.
+3. Monero's use of edwards25519 elliptic curve for its key pair generation, which limits our ability to use [adaptor signatures](https://github.com/LLFourn/one-time-VES/blob/master/main.pdf) with blockchains such as Bitcoin and Ethereum, which use a different curve.
+
+Thanks to the publication of [Joël Gugger's Bitcoin-Monero Cross-chain Atomic Swap paper](https://eprint.iacr.org/2020/1126.pdf) we discovered ways to overcome the afforementioned challenges.
+
+## Protocol
+
+At first we considered implementing the protocol as proposed by Joël, but we saw opportunities to simplify it, so we designed an alternate version.
+One of the changes was to reduce the number of transactions on Bitcoin by one.
+Thankfully, Joël and his team were kind enough to point out that we had introduced a race condition between the redeem and refund paths, caused by the non-instant finality of transactions on Bitcoin (and most other blockchains).
+
+Reverting said change left us with a largely similar protocol with some key differences:
+
+1. Simple 2-of-2 multisig scripts for all Bitcoin transactions.
+2. Transaction-level timelocks.
+3. An alternative approach to the DLEQ proof required to ensure that the correct keys are revealed when redeeming and refunding bitcoin.
+
+With (1) and (2) we wanted to make the Bitcoin scripts as simple as possible, to uphold the privacy properties of Monero.
+A 2-of-2 multisig is commonplace on Bitcoin, so it does not necessarily indicate that an atomic swap is taking place.
+The third point will be discussed after the protocol has been properly discussed in the next section.
+
+### Notation and terminology
+
+- When referring to secret keys we use lowercase e.g. `x`; when referring to public keys we use uppercase e.g. `X`.
+- We refer to Alice as the party that holds Monero and wants Bitcoin in exchange; we refer to Bob as the party that holds Bitcoin and wants Monero in exchange.
+
+### Assumptions
+
+We assume that Alice and Bob have exchanged a set of parameters before performing the protocol, namely:
+
+- the amounts being exchanged; and
+- the values of the timelocks used for two of the Bitcoin transactions, `t_1` and `t_2`, where `t_2 > t_1`.
```

Comment:
```suggestion
- the values of the timelocks used for two of the Bitcoin transactions, `t_1` and `t_2`.
```

This was a left over from the old protocol. If timelocks are relative this is not needed.